### PR TITLE
Fix golint errors in pkg/controller/job

### PR DESCRIPTION
**Pull request by avorima**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fixes golint errors in `pkg/controller/job`.

**Which issue(s) this PR fixes**:

Part of #68026

**Special notes for your reviewer**:

Due to golint's remarks I renamed the JobController to Controller. This may be a user-facing change depending on how you define users.

I ran unit tests against `./pkg/controllers/job` and integration tests against `./tests/integration/cronjob`.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -60,7 +60,8 @@ var (
 	MaxJobBackOff = 360 * time.Second
 )
 
-type JobController struct {
+// Controller is a controller for Jobs.
```
Comment (cheftako) [inline pkg/controller/job/job_controller.go]: Can we make this less redundant and more useful? Something like "Controller ensures that all Job objects have a corresponding Pod to perform that work."

File: pkg/controller/job/job_controller.go [hunk 2]
```diff
@@ -89,7 +90,8 @@ type JobController struct {
 	recorder record.EventRecorder
 }
 
-func NewJobController(podInformer coreinformers.PodInformer, jobInformer batchinformers.JobInformer, kubeClient clientset.Interface) *JobController {
+// NewController creates a new Controller.
```
Comment (cheftako) [inline pkg/controller/job/job_controller.go]: Again something more along the lines of "NewController returns a new job controller to keep the relevant pods in sync with the corresponding job objects"

File: pkg/controller/job/job_controller.go (hunk 1 above)
Comment (avorima) [reply pkg/controller/job/job_controller.go]: Yes, sorry these comments weren't very useful. I hope they're better now

**Review by cheftako — COMMENTED** [review]
/lgtm

**Pull request merged**